process/sync/shardblock.go

```diff
@@ -142,6 +142,78 @@ func NewShardBootstrap(
 	return &boot, nil
 }
 
+func (boot *ShardBootstrap) addHeaderToForkDetector(shardId uint32, nonce uint64, lastNotarizedMetaNonce uint64) {
+	header, headerHash, errNotCritical := boot.storageBootstrapper.getHeader(shardId, nonce)
+	if errNotCritical != nil {
+		log.Info(errNotCritical.Error())
+		return
+	}
+
+	shardHeader, ok := header.(*block.Header)
+	if !ok {
+		return
+	}
+
+	// search up from last notarized till shardHeader.Round
+	highestOwnHdrFromMetaChain := &block.Header{}
+	highestMetaNonceInStorer := lastNotarizedMetaNonce
+	for {
```

Comment:
This is so not ok :)

Reply:
it is a loop with breaks. It is a while. Go does not have while, only for.

Reply:
ok